CADRE/CADRE_group.py

```diff
@@ -0,0 +1,244 @@
+"""
+Group containing a single CADRE model.
+"""
+
+import numpy as np
+
+from openmdao.api import Group, IndepVarComp
+
+from CADRE.attitude import Attitude_Angular, Attitude_AngularRates, Attitude_Attitude, \
+    Attitude_Roll, Attitude_RotationMtx, Attitude_RotationMtxRates, Attitude_Torque
+from CADRE.battery import BatteryConstraints, BatteryPower, BatterySOC
+from CADRE.parameters import BsplineParameters
+from CADRE.comm import Comm_AntRotation, Comm_AntRotationMtx, Comm_BitRate, \
+    Comm_DataDownloaded, Comm_Distance, Comm_EarthsSpin, Comm_EarthsSpinMtx, \
+    Comm_GainPattern, Comm_GSposEarth, Comm_GSposECI, Comm_LOS, Comm_VectorAnt, \
+    Comm_VectorBody, Comm_VectorECI, Comm_VectorSpherical
+from CADRE.orbit import Orbit_Dynamics
+from CADRE.reactionwheel import ReactionWheel_Power, ReactionWheel_Torque, ReactionWheel_Dynamics
+from CADRE.solar import Solar_ExposedArea
+from CADRE.sun import Sun_LOS, Sun_PositionBody, Sun_PositionECI, Sun_PositionSpherical
+from CADRE.thermal_temperature import ThermalTemperature
+from CADRE.power import Power_CellVoltage, Power_SolarPower, Power_Total
+
+
+class CADRE(Group):
+    """
+    OpenMDAO implementation of the CADRE model.
+
+    Parameters
+    ----------
+    n : int
+        Number of time integration points.
+
+    m : int
+        Number of panels in fin discretization.
+
+    solar_raw1 : string
+        Name of file containing the angle map of solar exposed area.
+
+    solar_raw2 : string
+        Name of file containing the azimuth map of solar exposed area.
+
+    comm_raw : string
+        Name of file containing the az-el map of transmitter gain.
+
+    power_raw : string
+        Name of file containing a map of solar cell output voltage versus current,
+        area, and temperature.
+
+    initial_inputs : dict, optional
+        Dictionary of initial values for all parameters. Set the keys you want to
+        change.
+    """
+
+    def __init__(self, n, m, solar_raw1=None, solar_raw2=None, comm_raw=None,
+                 power_raw=None, initial_inputs=None):
```

Comment:
This stuff should be in `setup`, not init.

(Also, passed in parameters should go into the group's options dict.)

Reply:
neglected to ping you that I addressed this.. have been running MDP on the cluster hence the sibsequent changes to that script.. also set the seed for the comm deriv test that was occasionally failing with the random inputs.